core/src/state/world_state.rs

```diff
@@ -115,6 +125,31 @@ impl ClusterState {
     pub fn backend(&self, backend: &BackendId) -> Option<&BackendState> {
         self.backends.get(backend)
     }
+
+    pub fn locked(&self, lock: &str) -> Result<Option<BackendId>> {
+        let Some(lock_owner) = self.locks.get(lock) else {
+            // The lock does not exist.
+            return Ok(None)
+        };
+
+        let Some(backend) = self.backends.get(lock_owner) else {
+            // The lock exists, but the backend has been purged.
+            // This must be true, because an assignment message is the only way to create a lock.
+            return Ok(None)
```

Comment:
When can a backend get purged from the `self.backends` list? Is this an unexpected state? Should we log anything here?